crowbar_framework/lib/crowbar/backup/restore.rb

```diff
@@ -296,8 +297,12 @@ def proposal?(file)
         raw_data.key?("attributes") && raw_data.key?("deployment")
       end
 
-      def migrate_database(time)
-        Crowbar::Migrate.migrate!
+      def migrate_database(time, migration_level = nil)
+        if migration_level
+          Crowbar::Migrate.migrate_to(migration_level)
```

Comment:
Let me rephrase my point: if the migration level attribute in the backup is higher than what the rails app knows, then I think we should either do not accept to upload this backup or fail early when trying to restore the backup.

I hope it helps? (I'm a bit confused by how you split the options, but that's probably an issue in my brain ;-))